velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -198,5 +202,47 @@ TEST_F(DateTimeFunctionsTest, lastDay) {
   EXPECT_EQ(lastDayFunc(std::nullopt), std::nullopt);
 }
 
+TEST_F(DateTimeFunctionsTest, dateSub) {
+  const auto dateSubInt32 = [&](std::optional<int32_t> date,
+                                std::optional<int32_t> value) {
+    return evaluateOnce<int32_t, int32_t>(
+        "date_sub(c0, c1)", {date, value}, {DATE(), INTEGER()});
+  };
+  const auto dateSubInt16 = [&](std::optional<int32_t> date,
+                                std::optional<int16_t> value) {
+    return evaluateOnce<int32_t, int32_t>(
```

Comment:
Are these template parameters correct? This implementation looks exactly the same as dateSubInt32.

Reply:
The difference is on dateSub value type.

Reply:
I see. It is confusing to see `evaluateOnce<int32_t, int32_t>` in all these lambdas. It makes me think that inputs are always int32_t. Maybe change to `evaluateOnce<int32_t, int32_t, int8_t>` or `evaluateOnce<int32_t>`.

Reply:
Before implementation registered 3 date_sub functions by different input value types, this may introduce confusion and duplication. I reworked and keep only one function. Kindly help to review again? Thanks.